### PR TITLE
Refactor/remove group size

### LINKED ISSUE #253: Cut Group Size Parameter
# Description of the Issue
Given the priorities of the software project, cutting the group size parameter would help us ship 
a minimally viable product. 

## Add Detailed Comment with More Information
GatorGrouper currently accepts the number of groups or the size of groups as arguments for 
group creation. This is more convenient for the user, but given our current circumstance is not 
necessary. Cutting this function will allow us to simplify our code greatly.

### Assigned Developers
@paladp

### LINKED ISSUE #266: Without  `--num-group`, Using `--graph` Would Run Wrong Method `group_random`
@Michionlion It was there already, we didn't refactor it since it only uses `numgroup`.
```
elif (
            GG_ARGUMENTS.method == constants.ALGORITHM_GRAPH
            and GG_ARGUMENTS.num_group is not constants.DEFAULT_NUMGRP
        ):
            GROUPED_STUDENT_IDENTIFIERS = group_graph.group_graph_partition(
                SHUFFLED_STUDENT_IDENTIFIERS, GG_ARGUMENTS.num_group
            )
```
However, there was a bug that if we don't use `--num-group` with `--graph` and only use `--graph` the program will run `group_random`.  We can fix that bug with a different PR or just in [PR#261](https://github.com/GatorEducator/gatorgrouper/pull/261).

_Originally posted by @Lancasterwu in https://github.com/GatorEducator/gatorgrouper/pull/260#issuecomment-472910446_

### PR DESCRIPTION
**Pull request by paladp**
# Description of Pull Request

This pull request changes the group_creation, gatorgrouper_cli, and test files primarily in order to 
remove the group_size user option. Changes to group_creation remove the corresponding grouping methods and leave the names of the num_group methods unchanged. The changes to the gatorgrouper_cli serve to change the logic for selecting a grouping method to reflect the removal of the previous methods. Lastly, two test files were changed since they called the now deleted functions. Where possible, I tried to change the tests to the best of my ability to be able to work with the new group_creation methods. Where I could not change them or did not know whether I should, I commented them out and left a note in the comments. I'm not entirely confident in the test changes so help from @Lancasterwu and the rest of the testing team would be appreciated. 

Fixes #253 
Fixes #266 

## Type of Change

Please describe the pull request as one of the following: 

- [ ] Bug fix
- [X] Breaking change
- [x] New feature
- [ ] Documentation update
- [ ] Other 

## Tags
@aubreypc @Michionlion @Lancasterwu @huangs1 @Alex-Yarkosky

### CODE REVIEW

File: tests/test_group_method.py [hunk 1]
```diff
@@ -27,54 +27,46 @@ def test_group_random1():
...
+#     ]
+#     lst2 = ["Dan", "Angie", "Austin", "Izaak", "Nick", "Jeff"]
+#     size_count = group_size
+#     actual_output = group_creation.group_random_group_size(lst, group_size)
+#     actual_output2 = group_creation.group_random_group_size(lst2, group_size)
+#
+#     assert len(actual_output) == 12 // size_count
+#     assert len(actual_output[0]) == size_count
+#     assert len(actual_output2) == 6 // size_count
+#     assert len(actual_output2[0]) == size_count
```
Comment (Lancasterwu) [inline tests/test_group_method.py]: Is this suppose to be commented out?

File: tests/test_group_method.py [hunk 2]
```diff
@@ -87,32 +79,29 @@ def test_group_random_extra():
...
+#         ["Nick", True, False, True, False],
+#         ["Marvin", False, False, True, True],
+#         ["Evin", True, True, True, False],
+#         ["Nikki", True, True, False, False],
+#         ["Nick", True, False, True, False],
+#         ["Dan", False, True, False, True],
+#     ]
+#     returned_groups = group_creation.group_random_group_size(responses, grpsize)
+#     size_count = grpsize
+#     assert len(returned_groups[0]) == size_count
```
Comment (Lancasterwu) [inline tests/test_group_method.py]: What is the purpose of commenting all of code here?

File: tests/test_group_method.py (hunk 1 above)
Comment (paladp) [reply tests/test_group_method.py]: Where I could not change them or did not know whether I should, I commented them out and left a note in the comments

Comment (Lancasterwu) [reply tests/test_group_method.py]: Nvm, I didn't see that part in the description.

**Comment by Lancasterwu** [comment]
@Michionlion It does since graph algorithm only use and must use `--num-group`,

**Comment by Michionlion** [comment]
> @Michionlion It does since graph algorithm only use and must use `--num-group`,

From what I could see there is no path in `gatorgrouper_cli.py` that checks the method for graph.

**Comment by Lancasterwu** [comment]
@Michionlion It was there already, we didn't refactor it since it only uses `numgroup`.
```
elif (
            GG_ARGUMENTS.method == constants.ALGORITHM_GRAPH
            and GG_ARGUMENTS.num_group is not constants.DEFAULT_NUMGRP
        ):
            GROUPED_STUDENT_IDENTIFIERS = group_graph.group_graph_partition(
                SHUFFLED_STUDENT_IDENTIFIERS, GG_ARGUMENTS.num_group
            )
```
However, there was a bug that if we don't use `--num-group` with `--graph` and only use `--graph` the program will run `group_random`.  We can fix that bug with a different PR or just in [PR#261](https://github.com/GatorEducator/gatorgrouper/pull/261).

**Comment by paladp** [comment]
> @Michionlion It was there already, we didn't refactor it since it only uses `numgroup`.
> 
> ```
> elif (
>             GG_ARGUMENTS.method == constants.ALGORITHM_GRAPH
>             and GG_ARGUMENTS.num_group is not constants.DEFAULT_NUMGRP
>         ):
>             GROUPED_STUDENT_IDENTIFIERS = group_graph.group_graph_partition(
>                 SHUFFLED_STUDENT_IDENTIFIERS, GG_ARGUMENTS.num_group
>             )
> ```
> 
> However, there was a bug that if we don't use `--num-group` with `--graph` and only use `--graph` the program will run `group_random`. We can fix that bug with a different PR or just in [PR#261](https://github.com/GatorEducator/gatorgrouper/pull/261).

This has been fixed in the latest commit. Now, when I run 

```
pipenv run python3 gatorgrouper_cli.py --file students.csv --method graph
```

I get 

```
usage: gatorgrouper_cli.py [-h] [-d] [-v] --num-group NUM_GROUP --file FILE
                           [--method {graph,rrobin,random}]
                           [--absentees ABSENTEES [ABSENTEES ...]]
gatorgrouper_cli.py: error: the following arguments are required: --num-group

```
And running

```
pipenv run python3 gatorgrouper_cli.py --file students.csv --method graph --num-group 4
```

Gives us

```
GatorGrouper: Automatically Assign Students to Groups
https://github.com/GatorEducator/gatorgrouper

Group 1
student6
student1
student7

Group 2
student3
student12
student11

Group 3
student2
student8
student4

Group 4
student10
student5
student9
```` 

As expected

**Pull request merged**